src/script/miniscript.h

```diff
@@ -523,6 +523,20 @@ struct Node {
         }
     }
 
+    NodeRef<Key> Clone() const
+    {
+        // Use TreeEval() to avoid a stack-overflow due to recursion
+        auto upfn = [](const Node& node, Span<NodeRef<Key>> children) {
+            NodeRef<Key> ret(MakeNodeRef<Key>(node));
```

Comment:
Is there a reasonable way to do this without using the default copy constructor? Since node's destructor means using the implicit copy constructor is always dangerous, wouldn't it be best to delete it (if possible) as another reviewer suggested? (https://github.com/bitcoin/bitcoin/pull/30866#discussion_r1788683926)?

Reply:
Yes, done and deleted the copy constructor.